subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c

```diff
@@ -0,0 +1,1009 @@
+			~XCVR_TSM_TIMING38_SIGMA_DELTA_EN_TX_HI_MASK) +
+			XCVR_TSM_TIMING38_SIGMA_DELTA_EN_TX_HI(pll_dig_en+1);
+			/* sigma_delta_en */
+
+	XCVR_TSM->TIMING19   -= B1(WU_OPTIM); /* sy_pd_filter_charge_en */
+	XCVR_TSM->TIMING24   -= B1(WU_OPTIM); /* sy_divn_cal_en */
+	XCVR_TSM->TIMING13   -= B1(WU_OPTIM); /* sy_vco_autotune_en */
+	XCVR_TSM->TIMING17   -= B0(WU_OPTIM); /* sy_lo_tx_buf_en */
+	XCVR_TSM->TIMING26   -= B0(WU_OPTIM); /* tx_pa_en */
+	XCVR_TSM->TIMING35   -= B0(WU_OPTIM); /* tx_dig_en */
+	XCVR_TSM->TIMING14   -= B0(WU_OPTIM); /* sy_pd_cycle_slip_ld_ft_en */
+
+	XCVR_TSM->END_OF_SEQ -= B1(WU_OPTIM) + B0(WU_OPTIM);
+}
+#endif
+
+void radio_setup(void)
+{
+	XCVR_Init(GFSK_BT_0p5_h_0p5, DR_1MBPS);
+	XCVR_SetXtalTrim(41);
+
+#ifdef DISABLE_HPMCAL
+	hpmcal_disable();
+#endif
+
+	/* enable the CRC as it is disabled by default after reset */
+	XCVR_MISC->CRCW_CFG |= XCVR_CTRL_CRCW_CFG_CRCW_EN(1);
+
+	/* Assign Radio #0 Interrupt to GENERIC_FSK */
+	XCVR_MISC->XCVR_CTRL |= XCVR_CTRL_XCVR_CTRL_RADIO0_IRQ_SEL(3);
+
+	GENFSK->BITRATE = DR_1MBPS;
+
+	/*
+	 * Split the buffer in equal parts: first half for Tx,
+	 * second half for Rx
+	 */
+	GENFSK->PB_PARTITION = GENFSK_PB_PARTITION_PB_PARTITION(PB_RX);
+
+	/* Get warmpup times. They are used in TIFS calculations */
+	rx_wu = (XCVR_TSM->END_OF_SEQ & XCVR_TSM_END_OF_SEQ_END_OF_RX_WU_MASK)
+				>> XCVR_TSM_END_OF_SEQ_END_OF_RX_WU_SHIFT;
+	tx_wu = (XCVR_TSM->END_OF_SEQ & XCVR_TSM_END_OF_SEQ_END_OF_TX_WU_MASK)
+				>> XCVR_TSM_END_OF_SEQ_END_OF_TX_WU_SHIFT;
+
+	/* IRQ config, clear pending interrupts */
+	irq_disable(LL_RADIO_IRQn_2nd_lvl);
+	GENFSK->IRQ_CTRL = 0xffffffff;
+	GENFSK->IRQ_CTRL = GENFSK_IRQ_CTRL_GENERIC_FSK_IRQ_EN(1) |
+			   GENFSK_IRQ_CTRL_RX_WATERMARK_IRQ_EN(1) |
+			   GENFSK_IRQ_CTRL_TX_IRQ_EN(1) |
+			   GENFSK_IRQ_CTRL_T2_IRQ_EN(1);
+
+	/* Disable Rx recycle */
+	GENFSK->IRQ_CTRL |= GENFSK_IRQ_CTRL_CRC_IGNORE(1);
+	GENFSK->WHITEN_SZ_THR |= GENFSK_WHITEN_SZ_THR_REC_BAD_PKT(1);
+
+	get_isr_latency();
+}
+
+void radio_reset(void)
+{
+	irq_disable(LL_RADIO_IRQn_2nd_lvl);
+	/* Vega radio is never disabled therefore doesn't require resetting */
+}
+
+void radio_phy_set(u8_t phy, u8_t flags)
+{
+	ARG_UNUSED(phy);
+	ARG_UNUSED(flags);
+
+	/* This function should set one of three modes:
+	 * - BLE 1 Mbps
+	 * - BLE 2 Mbps
+	 * - Coded BLE
+	 * We set this on radio_setup() function. There radio is
+	 * setup for DataRate of 1 Mbps.
+	 * For now this function does nothing. In the future it
+	 * may have to reset the radio
+	 * to the 2 Mbps (the only other mode supported by Vega radio).
+	 */
+}
+
+void radio_tx_power_set(u32_t power)
+{
+	ARG_UNUSED(power);
+
+	/* tx_power_level should only have the values 1, 2, 4, 6, ... , 62.
+	 * Any value odd value (1, 3, ...) is not permitted and will result in
+	 * undefined behavior.
+	 * Those value have an associated db level that was not found in the
+	 * documentation.
+	 * Because of these inconsistencies for the moment this
```

Comment:
Seems this would be worth adding to the list of not yet supported features

Reply:
Right, will add this as well to the list of known limitations